flink-core/src/main/java/org/apache/flink/configuration/GlobalConfiguration.java

```diff
@@ -45,6 +45,11 @@ public final class GlobalConfiguration {
 
     private static final Logger LOG = LoggerFactory.getLogger(GlobalConfiguration.class);
 
+    /**
+     * @deprecated This file is deprecated and is only used to help users migrate their legacy
+     *     configuration files to the new configuration file `config.yaml`.
+     */
+    @Deprecated @VisibleForTesting
```

Comment:
How about to introduce a separate `ConfigurationFileMigrationUtils` class? It can be the only place to host logics which recognizes the legacy file.